rest-server/src/controller/user_controller.cpp

```diff
@@ -67,7 +67,24 @@ void UserController::HandlePost(http_request message) {
     }
 }
 void UserController::HandleDelete(http_request message) {
-    message.reply(status_codes::NotImplemented, ResponseNotImpl(methods::DEL));
+    try {
+        auto replyStatus = status_codes::OK;
+        auto path = message.relative_uri().to_string();
+        auto userID = ParseUserID(path);
+        if (userID.length() > 0) { //Verify request has user id
+            auto user = dao__.GetUserByID(std::stoi(userID)); //Gets user from DAO
```

Comment:
Right now, the ParseUserID does not verify that what is parsed is in fact an integer. This line here could cause the program to throw an error if the next part of the path contains any non-integer characters. To be fair, this should be handled in the ParseUserID function, it should spit back an empty string if not a valid user id (or an integer of value -1?).

Reply:
Right now it returns "400 Bad Request - stoi" but the server doesn't crash, so that's fine for now.